widgets/filter-feature-layer/src/setting/setting.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  SettingSection,
+  SettingRow
+} from "jimu-ui/advanced/setting-components";
+import { IMConfig } from "../config";
+import defaultI18nMessages from "./translations/default";
+
+const { useState, useEffect } = React;
+
+export default function (props: AllWidgetSettingProps<IMConfig>) {
+
+  const supportedDsTypes = Immutable([AllDataSourceTypes.FeatureLayer]);
+
+  const [fields, setFields] = useState([]);
+
+  // When the dataSource is changed, set the fields dropdown values.
+  useEffect(() => {
+    if (props.useDataSources && props.useDataSources.length === 1) {
+
+      const dsManager = DataSourceManager.getInstance();
+      const useDataSource = props.useDataSources[0];
+      const ds: FeatureLayerDataSource = dsManager.getDataSource(useDataSource.mainDataSourceId) as FeatureLayerDataSource;
+
+      const layerDefinition = ds.getLayerDefinition();
+      const fields = layerDefinition.fields.filter((fieldInfo) => {
+        return fieldInfo.type === 'esriFieldTypeString';
+      });
+      setFields(fields);
+    } else {
+      setFields([]);
+      props.onSettingChange({
+        id: props.id,
+        config: props.config.set('filterField', '')
+      });
+    }
+  }, [props.useDataSources]);
+
+  // When the dataSource is chosen, save it to the settings.
+  const onDataSourceChange = (useDataSources: UseDataSource[]) => {
+    if (!useDataSources) {
+      return
+    }
+
+    props.onSettingChange({
+      id: props.id,
+      useDataSources: useDataSources
+    });
+  }
+
+  // When a field is chosen from the dropdown, save it to the settings.
+  const fieldsListChangeHandler = (evt) => {
+    props.onSettingChange({
+      id: props.id,
+      config: props.config.set('filterField', evt.target.value)
+    });
+  }
+
+  return (
+    <div>
+      <div className="widget-setting-get-map-coordinates">
+
+        <SettingSection
+          className="data-source-selector-section"
+          title={props.intl.formatMessage({
+            id: "dataSourceSelectorLabel",
+            defaultMessage: defaultI18nMessages.selectDataSource
+          })}
+        >
+          <SettingRow>
+            <DataSourceSelector
+              types={supportedDsTypes}
+              useDataSourcesEnabled
+              mustUseDataSource
+              useDataSources={props.useDataSources}
+              onChange={onDataSourceChange} widgetId={props.id}
+            />
+
+          </SettingRow>
+        </SettingSection>
+
+        {props.useDataSources && props.useDataSources.length > 0 &&
+          <SettingSection
+            className="attribute-selector-section"
+            title={props.intl.formatMessage({
+              id: "chooseAttributeSelectorLabel",
+              defaultMessage: defaultI18nMessages.chooseAttribute
+            })}
+          >
+            <SettingRow>
+              <Select onChange={(e) => { fieldsListChangeHandler(e) }} value={props.config.filterField}>
```

Comment:
In the data source selector entry, there is a `FieldSelector` component as well.

Reply:
I did not know about `FieldSelector`, great to know that exists! Just updated to use it.